tests/basic/pure_test.go

```diff
@@ -2286,3 +2286,146 @@ func faLUNExists(faVolList []string, pvc string) bool {
+					AppKeys: Inst().AppList,
+				})
+				log.FailOnError(err, "Failed to schedule application of %v namespace", taskName)
+				contexts = append(contexts, context...)
+			}
+			ValidateApplications(contexts)
+		})
+
+		stepLog = "Get the pods from the created namespaces"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for _, context := range contexts {
+				err := createPodNodeMap(podNodeMap, context.App.NameSpace)
+				log.FailOnError(err, "Failed to create podNodeMap")
+			}
+		})
+
+		stepLog = "select the node with highest number of pods and cordon that node"
+		Step(stepLog, func() {
+			nodeToCordon, err = selectNode(podNodeMap)
+			log.FailOnError(err, "Failed to select node with highest number of pods")
+			log.InfoD("Node with highest number of pods: %v", nodeToCordon)
+			//cordon node with the highest number of pods
+			defer func() {
+				err = core.Instance().UnCordonNode(nodeToCordon, defaultCommandTimeout, defaultCommandRetry)
+				log.FailOnError(err, "Failed to uncordon node %v", nodeToCordon)
+				log.Infof("uncordoned node %v", nodeToCordon)
+			}()
+			err = core.Instance().CordonNode(nodeToCordon, defaultCommandTimeout, defaultCommandRetry)
+			log.FailOnError(err, "Failed to cordon node %v", nodeToCordon)
+			log.InfoD("cordoned node %v", nodeToCordon)
+			//delete pods from the cordoned node
+			for pod, node := range podNodeMap {
+				log.Infof("delete node name:%v , pod name:%v, %v", node["node"], pod, nodeToCordon)
+				if node["node"] == nodeToCordon {
+					err = core.Instance().DeletePod(pod, node["namespace"], true)
+					log.FailOnError(err, "Failed to delete pod %v", pod)
+					log.Infof("deleted pod:%v and namespace:%v", pod, node["namespace"])
+				}
+			}
+		})
+
+		stepLog = "Verify if pods are scheduled on other nodes"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//get the pods from the deleted namespace and check if they are scheduled on other nodes
+			for _, node := range podNodeMap {
+				pods, err := core.Instance().GetPods(node["namespace"], nil)
+				log.FailOnError(err, "Failed to get pods in namespace %v", node["namespace"])
+				for _, pod := range pods.Items {
+					log.Infof("Pod name: %v, Node name: %v, %v", pod.Name, pod.Spec.NodeName, nodeToCordon)
+					if pod.Spec.NodeName == nodeToCordon {
+						log.FailOnError(fmt.Errorf("pod %v is still running on node %v", pod.Name, nodeToCordon), "Pod should not be running on node %v", nodeToCordon)
+						log.InfoD("Pod :%v is running on node %v", pod.Name, pod.Spec.NodeName)
+					}
+				}
+			}
+		})
+		stepLog = "Destroy applications"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			var wg sync.WaitGroup
+			for _, node := range podNodeMap {
+				wg.Add(1)
+				go func(node map[string]string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					nameSpace := node["namespace"]
+					//delete namespace
+					err = core.Instance().DeleteNamespace(nameSpace)
```

Comment:
TearDownContext does not work?

Reply:
Since pods are deleted and new pods are coming up, teardown mostly won't work but i'll give it a try and see